controllers/application_poller_controller.go

```diff
@@ -79,7 +80,14 @@ func (p *Poller) Start(ch <-chan struct{}) error {
 	}
 	p.log.Info("Starting application poller")
 
-	ctx := context.Background()
+	ctx, cancel := context.WithCancel(context.Background())
+
+	// Handle controller-manager signal
+	go func() {
+		for range ch {
+			cancel()
+		}
```

Comment:
Just curious as to why this syntax instead of the `<- ch / cancel()`?

Reply:
Real reason, linter yelled at me.
But beyond that it's a bit more concise and simpler; since there's only a single channel we're working with, a for/select construct works but is a bit overkill.

Reply:
Right after I hit merge, I do wonder if I should have included a break in here. I'd imagine it's not super significant because this should only get triggered on shutdown when the manager notifies us of shutdown, and then we handle the ctx.Done() and return later on.